cmd/kops/create_cluster.go

```diff
@@ -354,6 +357,12 @@ func (c *CreateClusterCmd) Run(args []string) error {
 		fmt.Println("Previewing changes that will be made:\n")
 	}
 
+	if c.NoPublicIP {
+		fullCluster.Spec.AssociatePublicIP = fi.Bool(false);
```

Comment:
My first implementation before pull request was in such way (I look at your `--yes` parameter which is bool) and thought it would be great to implement it in a similar way.
Also in general by default all bool variables are false, so have `--associate-public-ip` brings simple name but will break backward compatibility which is not good.
So for me `--no-public-ip` looks for me more pretty and understandable for enabling/disabling some options and looks identical like `--yes`.

So maybe it would be great to rename it to `--no-associate-public-ip` and by default, it will be `true` and backward compatible.